hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/ha/EditLogTailer.java

```diff
@@ -172,14 +171,22 @@
    */
   private final long maxTxnsPerLock;
 
-  public EditLogTailer(FSNamesystem namesystem, Configuration conf) {
+  /**
+   * Timer instance to be set only using constructor.
+   * Only tests can reassign this by using setTimerForTests().
+   * For source code, this timer instance should be treated as final.
+   */
+  private Timer timer;
+
+  public EditLogTailer(FSNamesystem namesystem, Configuration conf,
+      Timer timer) {
```

Comment:
IMO it's better to add a new constructor to avoid changes in the calling classes:

```
public EditLogTailer(FSNamesystem namesystem, Configuration conf) {
  this(namesystem, conf, new Timer());
}

EditLogTailer(FSNamesystem namesystem, Configuration conf,
    Timer timer) {
  this.tailerThread = ...
  this.conf = conf;
  this.namesystem = namesystem;
  this.timer = timer;
  ...
}
```

But upon looking further, it seems that we never actually pass in a custom timer to the constructor; we only change it via `setTimer`, so I think we can get rid of the version that has a custom timer?